src/bench/logging.cpp

```diff
@@ -17,32 +17,62 @@ static void Logging(benchmark::Bench& bench, const std::vector<const char*>& ext
     bench.run([&] { log(); });
 }
 
-static void LoggingYoThreadNames(benchmark::Bench& bench)
+static void LogPrintLevelWithThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Error, "%s\n", "test"); });
 }
-static void LoggingNoThreadNames(benchmark::Bench& bench)
+
+static void LogPrintLevelWithoutThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Error, "%s\n", "test"); });
+}
+
+static void LogPrintWithCategory(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0", "-debug=net"}, [] { LogPrint(BCLog::VALIDATION, "%s\n", "test"); });
+}
+
+static void LogPrintWithoutCategory(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0", "-debug=0"}, [] { LogPrint(BCLog::VALIDATION, "%s\n", "test"); });
+}
+
+static void LogPrintfCategoryWithThreadNames(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintfCategory(BCLog::VALIDATION, "%s\n", "test"); });
+}
+
+static void LogPrintfCategoryWithoutThreadNames(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintfCategory(BCLog::VALIDATION, "%s\n", "test"); });
 }
-static void LoggingYoCategory(benchmark::Bench& bench)
+
+static void LogPrintfWithThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=0", "-debug=net"}, [] { LogPrint(BCLog::NET, "%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintf("%s\n", "test"); });
 }
-static void LoggingNoCategory(benchmark::Bench& bench)
+
+static void LogPrintfWithoutThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=0", "-debug=0"}, [] { LogPrint(BCLog::NET, "%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintf("%s\n", "test"); });
 }
-static void LoggingNoFile(benchmark::Bench& bench)
+
+static void LogWithoutDebugLogFile(benchmark::Bench& bench)
 {
     Logging(bench, {"-nodebuglogfile", "-debug=1"}, [] {
+        LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Error, "%s\n", "test");
+        LogPrint(BCLog::VALIDATION, "%s\n", "test");
+        LogPrintfCategory(BCLog::VALIDATION, "%s\n", "test");
         LogPrintf("%s\n", "test");
-        LogPrint(BCLog::NET, "%s\n", "test");
```

Comment:
The way this benchmark is implemented, we can identify a regression between code changes. But if we wanted to get a sense of how logging without a a debug.log file compares to logging with a debug.log file, this doesn't help, right? It should only log with one of {LogPrintLevel, LogPrint, LogPrintfCategory} in order to compare the result with one of the other benchmarks.

Reply:
That benchmark was added in fa31dc9b714 with the intention expressed in https://github.com/bitcoin/bitcoin/pull/18815#issuecomment-621404493, i.e. to test this code currently in `setup_common`:

```cpp
/**
 * Make a test setup that has disk access to the debug.log file disabled. Can
 * be used in "hot loops", for example fuzzing or benchmarking.
 */
template <class T = const BasicTestingSetup>
std::unique_ptr<T> MakeNoLogFileContext(const std::string& chain_name = CBaseChainParams::REGTEST, const std::vector<const char*>& extra_args = {})
{
    const std::vector<const char*> arguments = Cat(
        {
            "-nodebuglogfile",
            "-nodebug",
        },
        extra_args);

    return std::make_unique<T>(chain_name, arguments);
}
```
So it looks like the benchmark should be updated to pass `-nodebug` (and still bench with all macros passed here just in case). It also looks like the -debuglogfile help could be clearer. Updating.

Reply:
Thanks @kouloumos. Decided to add a comment to clarify the intention behind that -nodebuglogfile bench, and not add macros to it per your suggestion.